azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/model/AuthType.java

```diff
@@ -28,24 +30,30 @@ public static AuthType parseAuthType(String type) {
             return AUTO;
         }
         switch (type.toLowerCase().trim()) {
+            case "auto":
+                return AUTO;
+            case "service_principal":
+                return SERVICE_PRINCIPAL;
+            case "managed_identity":
+                return MANAGED_IDENTITY;
             case "azure_cli":
                 return AZURE_CLI;
-            case "intellij":
-                return INTELLIJ_IDEA;
             case "vscode":
                 return VSCODE;
+            case "intellij":
+                return INTELLIJ_IDEA;
+            case "azure_auth_maven_plugin":
+                return AZURE_AUTH_MAVEN_PLUGIN;
             case "device_code":
                 return DEVICE_CODE;
-            case "managed_identity":
-                return MANAGED_IDENTITY;
             case "oauth2":
                 return OAUTH2;
             case "visual_studio":
                 return VISUAL_STUDIO;
-            case "auto":
-                return AUTO;
+            default:
+                throw new UnsupportedOperationException(String.format("Invalid auth type '%s', supported values are: %s.", type,
```

Comment:
use `IllegalArgumentException` here?

Reply:
InvalidConfigurationException  will be better